### PR TITLE
Merge comp code for(from) week 0

### PR DESCRIPTION
**Pull request by S1ink**
Feel free to keep merging into this branch before w0

### CODE REVIEW

**Comment by S1ink** [comment]
You guys can review this now but we shouldn't merge it until after characterizing/more testing tomorrow night.

**Comment by beardedone55** [comment]
The following is just something to think about going forward.  I do not expect you to change all your code based on what I say next.

You use the `this` keyword a lot.  You don't have to use `this` to reference an **instance variable** (variables that are members of a class) unless the name of the **instance variable** you are referencing clashes with a method parameter name or a method local variable name.  Using the `this` keyword where it is not required adds a bit of clutter to the code.  (There may be others who disagree, and that's OK.  This is your project; you get to decide what's right.  Consistency is the important thing.)

As an example, the following code requires the use of the `this` keyword to assign a value to the **instance variable** named `x`, but does not require it for `y`:

```java
class Robot {
    private int x;
    private int y;

    public Robot(int x) {
        this.x = x; // 'this' keyword is required to differentiate 
                    // between the instance variable and the method parameter.
        y = 7;  // 'this' keyword is not required because instance variable 
                // does not share a name with another variable in this scope.
    }

}
```
If you are finding that you must use the `this` keyword, it may mean that you need to reconsider how you are naming your variables.  For example, in some projects, it is common practice to name all **instance variables** so that they begin with "m_" to make it clear that those variables are intended to be members of the class.

One other case where you must use the `this` keyword is when you want to call a constructor from another method (like another constructor).  This allows you to have multiple constructors that take different parameters but avoids duplicating code (which is good!).

Here is an example of using `this` to call another constructor (I see you also have examples of this in your robot code!).

```java
class Robot {
    private int m_x;
    private int m_y;

    public Robot(int x) {
        m_x = x;
        // Pretend a bunch of complex init code exists here...
        m_y = 42;
    }

    public Robot() {
        this(999);  //call other constructor with a default value of 999
    }
}
```

**Comment by S1ink** [comment]
@beardedone55 I am definitely aware of my overuse of the `this` keyword and that is just the way I like it to be. There are oftentimes a lot of large or nested classes, so I like to make it obvious where every variable is coming from so there is no confusion (and so I end up using it everywhere). Its a style I adopted a few years ago and I personally like it. I also do not like the convention of adding prefixes because it is sort of redundant (with java `static`, `final`, `private`/`public`, modifiers etc.) and I don't find them necessary. I guess I'm more open to using `m_` than I am `k`(constant here) - I really hate the `k` convention, its annoying and using all caps makes more sense and allows constants stand out way better. Either way, I totally agree that consistency is best and I have been working on being consistent with convention for a long time now.

File: src/main/java/frc/robot/Auto.java
Comment (beardedone55) [inline src/main/java/frc/robot/Auto.java:118]: Is there a reason why these instance variables are declared public?

**Review by Amelia1592 — APPROVED** [review]
This looks great!

File: src/main/java/frc/robot/Auto.java
Comment (S1ink) [reply src/main/java/frc/robot/Auto.java:118]: Those are static which means that they aren't instance variables - so yes there is a reason and that is because since they are final and meant to be constant values, I wanted them to be accessible outside the class if needed.

Comment (beardedone55) [reply src/main/java/frc/robot/Auto.java:118]: Ok, I missed that they are static; however, just from looking at them, it seems like these constants only make sense in the context of the implementation of the ClimbPad class.  Generally it's better to expose as little of the class's implementation as possible to other classes so that you can make changes to the implementation of the class without breaking other code that is using that class.

If you really need to access the constants from other classes, wouldn't it be better for them to be in the Constants class?

File: src/main/java/frc/robot/Constants.java
Comment (beardedone55) [inline src/main/java/frc/robot/Constants.java]: Are these the parameters you get out of the SysID tool you ran earlier today?

File: src/main/java/frc/robot/Constants.java
Comment (beardedone55) [inline src/main/java/frc/robot/Constants.java]: This semicolon looks lonely.

File: src/main/java/frc/robot/Constants.java
Comment (S1ink) [reply src/main/java/frc/robot/Constants.java]: Yes. I haven't pushed the new values yet though.

File: src/main/java/frc/robot/Constants.java
Comment (S1ink) [reply src/main/java/frc/robot/Constants.java]: He can stay, I will delete the one on the last full line instead.

File: src/main/java/frc/robot/Auto.java
Comment (S1ink) [reply src/main/java/frc/robot/Auto.java:118]: Yep, I just didn't want to add clutter if these values are only temporary (I'm not done with this routine). Once it is done they I will add the constants to the actual Constants.java file.

Comment (S1ink) [reply src/main/java/frc/robot/Auto.java:118]: Its also just so I can see what those values are easily and know for sure what they are for - if they were in constants I might forget if I was lazy with naming them.

File: src/main/java/frc/robot/Runtime.java
Comment (beardedone55) [inline src/main/java/frc/robot/Runtime.java]: Weren't you planning to use the velocity drive?

Comment (beardedone55) [reply src/main/java/frc/robot/Runtime.java]: Oh,  this is for arcade drive...

Comment (S1ink) [reply src/main/java/frc/robot/Runtime.java]: Yes I updated the controls in the latest commit (a few mins ago)

Comment (beardedone55) [reply src/main/java/frc/robot/Runtime.java]: Actually... I'm confused... this is tank drive.  What is the arcade board?  Is that the thing with the two flight sticks and the button box with the switches on top?

Comment (S1ink) [reply src/main/java/frc/robot/Runtime.java]: "Arcade Board" is just the 2 sticks, "Control Board" is that plus the buttonbox, and "Comp controls" are the 2 arcade sticks plus and xbox controller because the buttonbox was causing issues and that was the best thing I could come up with for tommorrow.

**Review by beardedone55 — APPROVED** [review]
Overall, looks good to me.

**Comment by S1ink** [comment]
Wow thanks, I didn't expect such an in-depth review. I'm still cleaning up some of the changes from yesterday, so once I push those I will merge this.

**Pull request merged**
